lib/inventory/metadata/metadata.go

```diff
@@ -210,14 +199,35 @@ func (c *fetchConfig) fetchContainerRuntime() string {
 
 // fetchContainerOrchestrator returns kubernetes-${GIT_VERSION} if the instance is
 // running on kubernetes.
+// This function performs the equivalent of the following:
+// curl -k https://$KUBERNETES_SERVICE_HOST:$KUBERNETES_SERVICE_PORT/version | jq .gitVersion
 func (c *fetchConfig) fetchContainerOrchestrator() string {
-	if c.kubeClient == nil {
+	url := fmt.Sprintf("https://%s:%s/version", c.getenv("KUBERNETES_SERVICE_HOST"), c.getenv("KUBERNETES_SERVICE_PORT"))
+	req, err := http.NewRequestWithContext(c.context, http.MethodGet, url, nil)
+	if err != nil {
+		log.Debugf("Failed to create kubernetes http GET request '%s': %s", url, err)
+		return ""
+	}
+
+	insecureSkipVerify := true
+	resp, err := c.httpDo(req, insecureSkipVerify)
+	if err != nil {
+		log.Debugf("Failed to perform http GET request: %s", err)
 		return ""
 	}
+	defer resp.Body.Close()
 
-	version, err := c.kubeClient.Discovery().ServerVersion()
+	body, err := io.ReadAll(resp.Body)
 	if err != nil {
-		log.Debugf("Failed to retrieve kubernetes server version: %s", err)
+		log.Debugf("Failed to read http response body: %s", err)
+		return ""
+	}
+
+	var version struct {
+		GitVersion string `json:"gitVersion"`
+	}
+	if err := json.Unmarshal(body, &version); err != nil {
+		log.Debugf("Failed to unmarshal http response body: %s", err)
 	}
 
 	return fmt.Sprintf("kubernetes-%s", version.GitVersion)
```

Comment:
Check if `version.GitVersion` is empty first?

Reply:
Done in cc98518e454d5cc8a14c6cf9a8500d56fbd63eab.